src/js/components/TextInput/TextInput.js

```diff
@@ -444,12 +469,31 @@ const TextInput = forwardRef(
 
     const textInputIcon = useSizedIcon(icon, rest.size, theme);
 
+    const CopyButton = (
```

Comment:
I assume this same component is also going to be used for MaskedInput and potentially TextArea wonder if we pull it out to its separate file and then import that way if we need to change anything it's in one place. just thought for future doesn't necessarily have to be in this PR

Reply:
Going to hold off consolidating until we add read only to more inputs, I was running into some issues trying to move the copy button to the utils/readonly file. It seems like there are problems importing grommet components with utils.

Reply:
thats fine!